client/consensus/grandpa/lib.go

```diff
@@ -20,10 +28,60 @@ type Authority[ID AuthorityID] struct {
 	Weight uint64
 }
 
+type AuthorityList[ID AuthorityID] []Authority[ID]
+
 // NewAuthoritySetStruct A new authority set along with the canonical block it changed at.
 type NewAuthoritySetStruct[H comparable, N constraints.Unsigned, ID AuthorityID] struct {
 	CanonNumber N
 	CanonHash   H
 	SetId       N
 	Authorities []Authority[ID]
 }
+
+type messageData[H comparable, N constraints.Unsigned] struct {
+	Round   uint64
+	SetID   uint64
+	Message finalityGrandpa.Message[H, N]
+}
+
+// Check a message signature by encoding the message as a localised payload and
+// verifying the provided signature using the expected authority id.
+// The encoding necessary to verify the signature will be done using the given
+// buffer, the original content of the buffer will be cleared.
+func checkMessageSignature[H comparable, N constraints.Unsigned, ID AuthorityID](
+	message any,
```

Comment:
why not just accept a `finalityGrandpa.Message` as the param instead of `any`?

Reply:
updated